ext/drx/scatter_gather_aarch64.c

```diff
@@ -366,46 +365,61 @@ expand_scalar_plus_vector(void *drcontext, instrlist_t *bb, instr_t *sg_instr,
-                /*disp=*/0, /*flags=*/0, sg_info->scalar_value_size,
-                sg_info->extend_amount);
-
+            /* ldr[bh]  scratch_gpr0, [base_reg, scratch_gpr0, mod #amount] */
             if (sg_info->is_scalar_value_signed) {
                 const reg_id_t ld_dst =
-                    reg_resize_to_opsz(scratch_gpr, sg_info->element_size);
+                    reg_resize_to_opsz(scratch_gpr0, sg_info->element_size);
                 switch (sg_info->scalar_value_size) {
                 case OPSZ_1: EMIT(ldrsb, opnd_create_reg(ld_dst), mem); break;
                 case OPSZ_2: EMIT(ldrsh, opnd_create_reg(ld_dst), mem); break;
                 case OPSZ_4: EMIT(ldrsw, opnd_create_reg(ld_dst), mem); break;
                 default: DR_ASSERT_MSG(false, "Invalid scatter_gather_info_t data");
                 }
             } else {
-                const reg_id_t scratch_gpr_w = reg_resize_to_opsz(scratch_gpr, OPSZ_4);
+                const reg_id_t scratch_gpr0_w = reg_resize_to_opsz(scratch_gpr0, OPSZ_4);
                 switch (sg_info->scalar_value_size) {
-                case OPSZ_1: EMIT(ldrb, opnd_create_reg(scratch_gpr_w), mem); break;
-                case OPSZ_2: EMIT(ldrh, opnd_create_reg(scratch_gpr_w), mem); break;
-                case OPSZ_4: EMIT(ldr, opnd_create_reg(scratch_gpr_w), mem); break;
-                case OPSZ_8: EMIT(ldr, opnd_create_reg(scratch_gpr), mem); break;
+                case OPSZ_1: EMIT(ldrb, opnd_create_reg(scratch_gpr0_w), mem); break;
+                case OPSZ_2: EMIT(ldrh, opnd_create_reg(scratch_gpr0_w), mem); break;
+                case OPSZ_4: EMIT(ldr, opnd_create_reg(scratch_gpr0_w), mem); break;
+                case OPSZ_8: EMIT(ldr, opnd_create_reg(scratch_gpr0), mem); break;
                 default: DR_ASSERT_MSG(false, "Invalid scatter_gather_info_t data");
                 }
             }
 
-            /* cpy      gather_dst_reg.element_size, scratch_pred/m, scratch_gpr */
-            EMIT(cpy_sve_pred,
-                 opnd_create_reg_element_vector(sg_info->gather_dst_reg,
-                                                sg_info->element_size),
-                 opnd_create_predicate_reg(scratch_pred, true),
-                 opnd_create_reg(reg_resize_to_opsz(scratch_gpr, sg_info->element_size)));
+            /* cpy      gather_dst_reg.element_size, scratch_pred/m, scratch_gpr0 */
+            EMIT(
+                cpy_sve_pred,
+                opnd_create_reg_element_vector(sg_info->gather_dst_reg,
+                                               sg_info->element_size),
+                opnd_create_predicate_reg(scratch_pred, true),
+                opnd_create_reg(reg_resize_to_opsz(scratch_gpr0, sg_info->element_size)));
         } else {
-            DR_ASSERT_MSG(sg_info->is_load, "Stores are not yet supported");
+            DR_ASSERT_MSG(!sg_info->is_scalar_value_signed,
```

Comment:
Does this mean there are no scatter instrs where the value is signed?

Reply:
Yes. The same is true for scalar stores. Load instructions need to extend the read value to the register size so there are unsigned and signed load instructions to control whether it zero extends or sign extends it. There is no extension happening with a store so we don't need signed/unsigned variants.